torch/_inductor/codegen/cuda/cutlass_python_evt.py

```diff
@@ -0,0 +1,201 @@
+
+
+class _ComputeAssignmentFormatter(DefaultHandler):
+    def __init__(self, parent_handler: Any):
+        super().__init__()
+        self.parent_handler = parent_handler
+
+    def _default(self, name: str, args: tuple[Any, ...], kwargs: dict[str, Any]) -> Any:
+        if name.startswith("_"):
+            raise NotImplementedError(name)
+
+        if hasattr(self.parent_handler, f"{name}"):
+            fargs = [_arg_str(a) for a in args]
+            fkwargs = {key: _arg_str(a) for key, a in kwargs.items()}
+            fn = getattr(self.parent_handler, f"{name}")
+            line = fn(*fargs, **fkwargs)
+            if name in ("load", "store"):
+                return line
+            else:
+                var = self.parent_handler._tmp_var()
+                self.parent_handler.output.writeline(f"{var} = {line}")
+            return var
+        else:
+            raise NotImplementedError(name)
+
+
+class CutlassEVTCodegen:
+    """
+    Notes:
+        * Used by CUTLASSGemmTemplate.
+        * This class should not be instantiated by users, it is intended to be used
+            by calling CutlassEVTCodegen.ir_to_evt_python_code(...)
+            which instantiates this class as an ops handler for virtualized.V.ops.[op-name]
+        * Extend this with more _op_<whatever> nodes to add support for new pointwise operations.
+    """
+
+    def __init__(self, accumulator_node_name: str):
+        """
+
+        Initializes a CutlassEVTEpilogueArgumentFormatter object. Do not instantiate directly.
+        Use the CutlassEVTCodegen.ir_to_evt_python_code static method.
+
+        Args:
+            accumulator_node_name (str): The name of the accumulator node which should contain
+                                          the Matmul result before fusion according to the IR graph.
+        """
+        self.accumulator_node_name: str = accumulator_node_name  #
+        self.output: IndentedBuffer = IndentedBuffer(1)  # The output buffer for codegen
+        self.var_counter: int = (
+            0  # used to generate variable names, incremented for each new variable
+        )
+        self.result_aliases: dict[str, OpsValue] = (
+            dict()
+        )  # Aliases for subexpression functors
+        self.reads: OrderedSet[str] = OrderedSet()
+        self.writes: OrderedSet[str] = OrderedSet()
+
+    @staticmethod
+    def ir_to_evt_python_code(
+        cuda_template_node_name: str,
+        epilogue_nodes: list[IRNode],
+    ) -> tuple[list[str], list[str], dict[str, Any], str]:
+        codegen = CutlassEVTCodegen(cuda_template_node_name)
+        handler = _ComputeAssignmentFormatter(codegen)
+
+        D_defined = False
+
+        with (
+            virtualized.V.set_ops_handler(handler),
+            patch.object(  # type: ignore[call-arg]
+                FlexibleLayout, "allow_indexing", True
+            ),
+        ):
+            for node in epilogue_nodes:
+                assert isinstance(node, ComputedBuffer)
```

Comment:
Where do we handle upcasting the mm output to fp32 ? / making sure epilogue is done in fp32 ?

Reply:
I was able to compile mixed dtype epilogues so it will respect whatever dtype the epilogue nodes have.